compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3423,9 +3661,31 @@ TR::Register *OMR::X86::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::Co
          elementSize = TR::Symbol::convertTypeToSize(dt);
       }
 
-   bool isShortConstantArray = false;
-   if (isShortConstantArray)
+   bool isShortConstArrayWithDirection = false;
+   bool isShortConstArrayWithoutDirection = false;
+   uint32_t size;
+   if (sizeNode->getOpCode().isLoadConst() && TR::Compiler->target.is64Bit())
+      {
+      size = TR::TreeEvaluator::integerConstNodeValue(sizeNode, cg);
+      if (node->isForwardArrayCopy() || node->isBackwardArrayCopy())
+         {
+         if (size <= 256) isShortConstArrayWithDirection = true;
+         }
+      else
+         {
+         if (size <= 16*4) isShortConstArrayWithoutDirection = true;
```

Comment:
I would much prefer consts/static consts to #defines - makes it way easier for the compiler and other tools to deal with the constants as what they are rather than magic token pasting from the preprocessor which is evil.